concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java

```diff
@@ -62,6 +64,61 @@ public Iterator<AnonymityRevokerInfo> getAnonymityRevokers(final BlockHashInput
         return to(grpcOutput, ClientV2MapperExtensions::to);
     }
 
+    /**
+     * Gets an {@link Iterator} of Blocks Arriving at the node.
+     * With Connection Timeout.
+     * Form the time request is made and onwards.
+     * This can be used to listen for incoming blocks.
+     *
+     * @return {@link Iterator< BlockInfoV2 >}
+     */
+    public Iterator<BlockInfoV2> getBlocks() {
+        return this.getBlocks(timeout);
+    }
+
+    /**
+     * Gets an {@link Iterator} of Blocks Arriving at the node.
+     * With Specified Timeout.
+     * Form the time request is made and onwards.
+     * This can be used to listen for incoming blocks.
+     *
+     * @param timeoutMillis Timeout for the request in Milliseconds.
+     * @return {@link Iterator< BlockInfoV2 >}
+     */
+    public Iterator<BlockInfoV2> getBlocks(int timeoutMillis) {
+        var grpcOutput = this.server(timeoutMillis).getBlocks(Empty.newBuilder().build());
+
+        return to(grpcOutput, ClientV2MapperExtensions::to);
+    }
+
+    /**
+     * Gets an {@link Iterator} of Finalized Blocks.
+     * With Connection Timeout.
+     * Form the time request is made and onwards.
+     * This can be used to listen for blocks being Finalized.
+     *
+     * @return {@link Iterator< BlockInfoV2 >}
+     */
+    public Iterator<BlockInfoV2> getFinalizedBlocks() {
+        return this.getFinalizedBlocks(timeout);
+    }
+
+    /**
+     * Gets an {@link Iterator} of Finalized Blocks.
+     * With Specified Timeout.
+     * Form the time request is made and onwards.
+     * This can be used to listen for blocks being Finalized.
+     *
+     * @param timeoutMillis Timeout for the request in Milliseconds.
+     * @return {@link Iterator< BlockInfoV2 >}
+     */
+    public Iterator<BlockInfoV2> getFinalizedBlocks(int timeoutMillis) {
```

Comment:
Timeout has been included so that it can be overridden per request. 
That should be needed here because same client can be used to issue multiple requests and this method is specifically documented to be listening to incoming finalised blocks